### PR TITLE
windows/capsule: Update InfGenerator with integrity file

### PR DESCRIPTION
**Pull request by corthon**
Add the ability to set the IntegrityFilename and populate the data in the produced INF file.

Also refactored the way some of the INF sections are produced to make it easier to dynamically
add entries to these sections.

### CODE REVIEW

**Comment by corthon** [comment]
Need to confirm that this produces the exact same INF as the old version.

File: edk2toollib/windows/capsule/inf_generator.py [hunk 1]
```diff
@@ -14,6 +14,15 @@
 import uuid
 
 
+class InfSection(object):
+    def __init__(self, name) -> None:
+        self.Name = name
+        self.Items = []
+
+    def __str__(self) -> str:
```
Comment (spbrogan) [inline edk2toollib/windows/capsule/inf_generator.py:22]: can you put a trivial unit test in place for this?

Comment (corthon) [reply edk2toollib/windows/capsule/inf_generator.py:22]: Done.

**Review by spbrogan — CHANGES_REQUESTED** [review]

File: edk2toollib/windows/capsule/inf_generator.py [hunk 2]
```diff
@@ -40,22 +49,17 @@ class InfGenerator(object):
 [Firmware_Install.NT]
 CopyFiles = Firmware_CopyFiles
 {Rollback}
-[Firmware_CopyFiles]
-{FirmwareBinFile}
+{FirmwareCopyFilesSection}
```
[the remark this answers is not in the document]
Comment (corthon) [reply edk2toollib/windows/capsule/inf_generator.py:52]: From the description:
"Also refactored the way some of the INF sections are produced to make it easier to dynamically
add entries to these sections."

Yeah, it's to make them manipulable.

File: edk2toollib/windows/capsule/inf_generator.py [hunk 3]
```diff
@@ -74,10 +78,10 @@ class InfGenerator(object):
 
     ROLLBACKTEMPLATE = r"""AddReg    = Firmware_DowngradePolicy_Addreg
 
-  ;override firmware resource update policy to allow downgrade to lower version
-  [Firmware_DowngradePolicy_Addreg]
-  HKLM,SYSTEM\CurrentControlSet\Control\FirmwareResources\{{{EsrtGuid}}},Policy,%REG_DWORD%,1
-  """
+;override firmware resource update policy to allow downgrade to lower version
+[Firmware_DowngradePolicy_Addreg]
```
[the remark this answers is not in the document]
Comment (corthon) [reply edk2toollib/windows/capsule/inf_generator.py:82]: I thought about that... but I never had a need to modify it dynamically. Still could, if it's better for consistency.

**Review by spbrogan — APPROVED** [review]

File: edk2toollib/windows/capsule/inf_generator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (corthon) [reply edk2toollib/windows/capsule/inf_generator.py:52]: Yes.

**Pull request merged**